http-server/src/main/java/io/airlift/http/server/HttpServer.java

```diff
@@ -177,6 +179,11 @@ public HttpServer(
             baseHttpConfiguration.setResponseHeaderSize(toIntExact(config.getMaxResponseHeaderSize().toBytes()));
         }
 
+        if (legacyUriComplianceEnabled) {
+            // allow encoded slashes to occur in URI paths
+            baseHttpConfiguration.setUriCompliance(UriCompliance.LEGACY);
```

Comment:
Don't use LEGACY.

You DEFINITELY do not want to enable all of the other things that LEGACY also brings to the table.  Namely: AMBIGUOUS_EMPTY_SEGMENT, AMBIGUOUS_PATH_ENCODING, AMBIGUOUS_PATH_PARAMETER, and AMBIGUOUS_PATH_SEGMENT.

Be surgical, only turn on what you need/want to allow, not everything.

Eg:

``` java
UriCompliance uriCompliance = UriCompliance.from("RFC3986,AMBIGUOUS_PATH_SEPARATOR");
baseHttpConfiguration.setUriCompliance(uriCompliance);
```

or 

``` java
UriCompliance uriCompliance = UriCompliance.from(EnumSet.of(UriCompliance.Violation.AMBIGUOUS_PATH_SEPARATOR));
baseHttpConfiguration.setUriCompliance(uriCompliance);
```

Reply:
Thanks for this. It wasn't obvious how to construct that.